libraries/standard/http/include/http_client.h

```diff
@@ -0,0 +1,339 @@
+
+    /**
+     * @brief Optional callback for intercepting the header during the first 
+     * parse through of the response as is it receive from the network.
+     */
+    httpclientHeaderParsingCallback_t* pHeaderParsingCallback;
+    
+    /**
+     * @brief The Location of the response headers in pBuffer.
+     * 
+     * This is updated by #HTTPClient_Send.
+     */
+    uint8_t* pHeaders;
+
+    /**
+     * @brief Byte length of the response headers in pBuf.
+     * 
+     * This is updated by #HTTPClient_Send.
+     */
+    size_t headersLen; 
+
+    /**
+     * @brief The location of the response body in pBuf.
+     * 
+     * This is updated by #HTTPClient_Send.
+     */
+    uint8_t* pBody;
+
+    /**
+     * @brief Byte length of the body in pBuf.
+     * 
+     * This is updated by #HTTPClient_Send.
+     */
+    size_t bodyLen;
+    
+    /* Useful HTTP header values found. */
+
+    /**
+     * @brief The HTTP response Status-Code.
+     * 
+     * This is updated by #HTTPClient_Send.
+     */
+    uint16_t statusCode;
+    
+    /**
+     * @brief The value in the "Content-Length" header is returned here.
+     * 
+     * This is updated by #HTTPClient_Send.
+     */
+    size_t contentLength;
+
+    /**
+     * @brief Count of the headers sent by the server.
+     * 
+     * This is updated by #HTTPClient_Send.
+     */
+    size_t headerCount;
+
+    /**
+     * @brief Flags of useful headers found in the response.
+     * 
+     * This is updated by #HTTPClient_Send.
+     */
+    uint32_t flags;
+} HTTPResponse_t;
+
+
+/**
+ * TODO: Add through documentation.
+ */
+HTTPStatus_t HTTPClient_InitializeRequestHeaders( HTTPRequestHeaders_t * pRequestHeaders, 
+                                                      const HTTPRequestInfo_t * pRequestInfo );
+
+/**
+ * TODO: Add thorough documentation.
+ */
+HTTPStatus_t HTTPClient_AddHeader( HTTPRequestHeaders_t * pRequestHeaders, 
+                                       const char* pName, 
+                                       size_t nameLen, 
+                                       const char * pValue, 
+                                       size_t valueLen );
+
+/**
+ * TODO: Add thorough documentation.
+ */
+HTTPStatus_t HTTPClient_AddRangeHeader( HTTPRequestHeaders_t *pRequestHeaders,
+                                            int32_t rangeStart,
+                                            int32_t rangeEnd );
+
+/**
+ * TODO: Add thorough documentation.
+ */
+HTTPStatus_t HTTPClient_Send( const HTTPTransportInterface_t* pTransport,
+                                  const HTTPRequestHeaders_t* pRequestHeaders,
+                                  const uint8_t* pRequestBodyBuf, // For a PUT or POST request.
+                                  size_t reqBodyBufLen,
+                                  HTTPResponse_t* pResponse );
+
+/**
+ * TODO: Add thorough documentation.
+ */
+HTTPStatus_t HTTPClient_ReadHeader( HTTPResponse_t* pResponse,
```

Comment:
The `pResponse` input parameter can be of `const` type

Reply:
Thanks